src/Service/SchemaGenerator.php

```diff
@@ -32,27 +32,39 @@ public function getSchema(Member $member)
     {
         $enforcementManager = EnforcementManager::singleton();
 
+        $schema = self::getMethodSchema($member) + [
+            'canSkip' => $enforcementManager->canSkipMFA($member),
+            'isFullyRegistered' => $enforcementManager->hasCompletedRegistration($member),
+            'resources' => $this->getResources(),
+            'shouldRedirect' => $enforcementManager->shouldRedirectToMFA($member),
+        ];
+
+        $this->extend('updateSchema', $schema);
+
+        return $schema;
+    }
+
+    /**
+     * Provides schema related to MFA methods, using the current Member as context
+     *
+     * @param Member&MemberExtension $member
+     * @return array
+     */
+    public function getMethodSchema(Member $member)
```

Comment:
I'm still not sure about this change. It's only text, I'm still in the keep it simple camp

Reply:
It's an attempt to keep a bunch of unnecessary config from being dumped in the field element. I _could_ just pull out the relevant pieces in the Field class, though. 🤔

Reply:
Done 👍